app/test5.py

```diff
@@ -0,0 +1,20 @@
+def openai_api_call(prompt_text):
+    import openai
+    from os import getenv
+    from dotenv import load_dotenv
+
+    load_dotenv()
+
+    api_key = getenv('OPENAI_API_KEY')
+    if not api_key:
+        raise EnvironmentError("OPENAI_API_KEY is not set in the environment variables.")
+    openai.api_key = api_key
+    try:
+        response = openai.Completion.create(
+            engine="gpt-4",
+            prompt=prompt_text,
+            max_tokens=150
+        )
+        return response.choices[0].text.strip()
+    except Exception as e:
+        return f"An error occurred: {str(e)}"
```

Comment:
Directly returning error messages as part of normal program flow is not a best practice, as it can lead to mixing error handling logic with business logic. Consider raising an exception or using a result object to encapsulate success/failure states.

**Label**: best practice

**Code suggestion**

```python
except Exception as e:
    raise SomeCustomException(f"An error occurred: {str(e)}")
```